content/en/user-guide/integrations/sdks/java/index.md

```diff
@@ -0,0 +1,245 @@
+---
+title: "Java"
+categories: []
+tags: ["sdk"]
+description: >
+    How to use the AWS Java SDK with LocalStack.
+aliases:
+- /integrations/sdks/java/
+---
+
+## Overview
+
+The AWS SDK for Java provides a Java API for AWS services. Using the SDK, your Java application can interact
+with LocalStack services the same way it does with Amazon services. Support for new services is regularly added to
+the SDK. For a list of the supported services and their API versions that are 
+included with each release of the SDK, view the [release notes](https://github.com/aws/aws-sdk-java#release-notes) 
+for the version that you’re working with.
+
+The Java SDK currently supports two major versions:
+
+* [AWS SDK for Java v1](https://github.com/aws/aws-sdk-java)
+* [AWS SDK for Java v2](https://github.com/aws/aws-sdk-java-v2)
+
+## Examples
+
+Full examples for both SDK versions can be found in the 
+[example repository](https://github.com/localstack/localstack-aws-sdk-examples/tree/main/java). This includes proper
+exception handling and all the necessary Maven dependencies.
+The scripts to create the AWS services on LocalStack can be found under the `src/main/resources` folder
+of each module in the repository. 
+
+Below you'll find an example of how to create an S3 client with the endpoint configured for LocalStack.
+The client can be used to upload a file to an existing bucket and then retrieve it.
+
+{{< tabpane >}}
+{{< tab header="v1" lang="java" >}}
+
+    // Credentials that can be replaced with real AWS values. (To be handled properly and not hardcoded.)
+final String ACCESS_KEY = "test";
+final String SECRET_KEY = "test";
+
+    // Existing bucket name.
+final String BUCKET_NAME = "records";
+
+    // The key of the object in the bucket, usually the name of the file.
+final String key = "hello-v1.txt";
+
+    // S3 Client with configured credentials, endpoint directing to LocalStack and desired region.
+AmazonS3 s3Client = AmazonS3ClientBuilder.standard()
+.withCredentials(new AWSStaticCredentialsProvider(credentials))
+.withEndpointConfiguration(new EndpointConfiguration("s3.localhost.localstack.cloud:4566",
+Regions.US_EAST_1.getName()))
+.build();
```

Comment:
```suggestion
AmazonS3 s3Client = AmazonS3ClientBuilder.standard()
  .withCredentials(new AWSStaticCredentialsProvider(credentials))
  .withEndpointConfiguration(new EndpointConfiguration("s3.localhost.localstack.cloud:4566", Regions.US_EAST_1.getName()))
  .build();
```

indenting the function chaining makes the code a bit easier to read. maybe we could add this everywhere

Reply:
I added the indentation and separation of client configuration and interaction with the services. I also made a clearer separation of sections based on services. As for the client configs, there are some options to add special handlers and some monitoring but that is really niche and should be covered in the official docs.
I think the configs in general are pretty self-explanatory 🤔